packages/@react-aria/table/src/useTableColumnResize.ts

```diff
@@ -76,45 +81,108 @@ export function useTableColumnResize<T>(props: AriaTableColumnResizeProps<T>, st
+    max = Number.MAX_SAFE_INTEGER;
+  }
+  let value = Math.floor(stateRef.current.getColumnWidth(item.key));
 
   let ariaProps = {
-    role: 'separator',
     'aria-label': props.label,
-    'aria-orientation': 'vertical',
-    'aria-labelledby': item.key,
-    'aria-valuenow': stateRef.current.getColumnWidth(item.key),
-    'aria-valuemin': stateRef.current.getColumnMinWidth(item.key),
-    'aria-valuemax': stateRef.current.getColumnMaxWidth(item.key)
+    'aria-orientation': 'horizontal' as 'horizontal',
+    'aria-labelledby': `${id} ${getColumnHeaderId(state, item.key)}`,
+    'aria-valuetext': stringFormatter.format('columnSize', {value}),
+    min,
+    max,
+    value
+  };
+
+  const focusInput = useCallback(() => {
+    if (ref.current) {
+      focusWithoutScrolling(ref.current);
+    }
+  }, [ref]);
+
+  let onChange = (e: ChangeEvent<HTMLInputElement>) => {
+    let currentWidth = stateRef.current.getColumnWidth(item.key);
+    let nextValue = parseFloat(e.target.value);
+
+    if (nextValue > currentWidth) {
+      nextValue = currentWidth + 10;
+    } else {
+      nextValue = currentWidth - 10;
+    }
+    stateRef.current.onColumnResize(item, nextValue);
+  };
+
+  let onDown = () => {
+    focusInput();
+    addGlobalListener(window, 'mouseup', onPointerUp, false);
+    addGlobalListener(window, 'touchend', onUp, false);
+    addGlobalListener(window, 'pointerup', onPointerUp, false);
+  };
+
+  let onPointerUp = (e) => {
+    // don't hide the resizer for touch since it's harder to bring back
+    if (e.pointerType === 'touch') {
+      focusInput();
+    } else {
+      focusSafely(triggerRef.current);
+    }
+    removeGlobalListener(window, 'mouseup', onPointerUp, false);
+    removeGlobalListener(window, 'touchend', onUp, false);
+    removeGlobalListener(window, 'pointerup', onPointerUp, false);
+  };
+
+  let onUp = () => {
+    focusInput();
+    removeGlobalListener(window, 'mouseup', onPointerUp, false);
+    removeGlobalListener(window, 'touchend', onUp, false);
+    removeGlobalListener(window, 'pointerup', onPointerUp, false);
   };
 
   return {
-    resizerProps: {
-      ...mergeProps(
-        moveProps,
-        {
-          onFocus: () => {
-            // useMove calls onMoveStart for every keypress, but we want resize start to only be called when we start resize mode
-            // call instead during focus and blur
-            stateRef.current.onColumnResizeStart(item);
-            state.setKeyboardNavigationDisabled(true);
-          },
-          onBlur: () => {
-            stateRef.current.onColumnResizeEnd(item);
-            state.setKeyboardNavigationDisabled(false);
-          },
-          tabIndex: showResizer ? 0 : undefined
+    resizerProps: mergeProps(
+      keyboardProps,
+      moveProps,
+      {
+        onMouseDown: (e: React.MouseEvent<HTMLElement>) => {
+          if (e.button !== 0 || e.altKey || e.ctrlKey || e.metaKey) {
+            return;
+          }
+          onDown();
```

Comment:
this is copied from the slider code
which is track listener code, so this may not be necessary? https://github.com/adobe/react-spectrum/blob/main/packages/%40react-aria/slider/src/useSlider.ts#L198
I'll double check

Reply:
sorry i linked the wrong one, ignore my previous comment, it's from slider thumb
https://github.com/adobe/react-spectrum/blob/main/packages/%40react-aria/slider/src/useSliderThumb.ts#L212 which makes more sense
I've switched to usePress and it appears to work just fine. It may just be old code that needed altKey etc before that existed in the PressEvent